### ISSUE TITLE
Hacktivate option not working on 3G

### ISSUE THREAD

**Issue by circularsprojects** [LukeZGD/Legacy-iOS-Kit#443]
Trying to hacktivate my iPhone 3G, but ideviceactivation isn't reporting the device as "SIM Required", even though the phone is displaying "Insert a valid SIM with no PIN lock to activate iPhone"

```
*** Legacy iOS Kit ***
 - Script by LukeZGD -

* Version: v24.03.29 (d2550c1)
* Platform: macos (14.1.1) 

* Device: iPhone1,2 (n82ap) in Normal mode
[WARNING] This device is only partially supported by Legacy iOS Kit. Some features may not work properly.
* iOS Version: 4.2.1 (8C148)
* ECID: 284039486837

 > Main Menu > Other Utilities
[Input] Select an option:
1) SSH Ramdisk		       3) Hacktivate Device	      5) Restart Device	     7) Enter DFU Mode		    9) (Re-)Install Dependencies
2) Attempt Activation	       4) Shutdown Device	      6) Enter Recovery Mode	     8) Create Custom IPSW	   10) Go Back
#? 3
[Log] Checking ideviceactivation status...
[WARNING] The SIM Required message did not show up at ideviceactivation, cannot continue.

* Save the terminal output now if needed.
* Legacy iOS Kit v24.03.29 (d2550c1)
* Platform: macos (14.1.1) 
```

Attempting activation doesnt work either, but I think thats just because I don't have an actual SIM to use with it.

**Comment by LukeZGD** [comment]
What's the error that shows up when attempting to activate with "Attempt Activation"? 

So I can change the "SIM Required" check

**Comment by circularsprojects** [comment]
> What's the error that shows up when attempting to activate with "Attempt Activation"? So I can change the "SIM Required" check

Activation server reports errors.
Activation Error

> Also is the device already jailbroken with the "Jailbreak Device" option? Since that is needed before hacktivation

When I try to jailbreak the device, it gets as far as attempting to SSH into the device, before failing
[3gjb.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14661303/3gjb.txt)

**Comment by LukeZGD** [comment]
Thanks for letting me know and sharing the log. From what I can see, it might be because I'm sending the files to the device too fast that it can't keep up since it's old and all...

Here's a test build that adds a delay for the jailbreak device option, as well as removing the ideviceactivation check: https://github.com/LukeZGD/Legacy-iOS-Kit/archive/refs/heads/test.zip

**Comment by LukeZGD** [comment]
Thanks for letting me know and sharing the log. From what I can see, it might be because I'm sending the files to the device too fast that it can't keep up since it's old and all...

Here's a test build that adds a delay for the jailbreak device option, as well as removing the ideviceactivation check: https://github.com/LukeZGD/Legacy-iOS-Kit/archive/refs/heads/test.zip

**Comment by circularsprojects** [comment]
[3gjbt.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14661535/3gjbt.txt)
I'm getting the same issue trying to jailbreak the device.

**Comment by LukeZGD** [comment]
What is happening lol, I don't get why it doesn't work... Please do a restore to 4.2.1 again and get the full log/output of idevicerestore, I think there is just something I'm not getting here

**Comment by circularsprojects** [comment]
[3grestore.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14661782/3grestore.txt)
restored to 4.2.1

[3gjbrestore.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14661785/3gjbrestore.txt)
tried to jb again just in case but still same issue

**Comment by LukeZGD** [comment]
Here's another test build (002test): https://github.com/LukeZGD/Legacy-iOS-Kit/archive/refs/heads/test.zip

Same link but it's updated, use this build to attempt the jailbreak

Sorry about this btw, support for the 2G, 3G, and touch 1 devices is currently in a broken state, but I'll do what I can to fix it

**Comment by circularsprojects** [comment]
[3g003.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14662290/3g003.txt)
Same issue, but this time instead of the phone being left on the recovery screen, it's left on a white screen

**Comment by LukeZGD** [comment]
Run `irecovery -s` while the device is in the white screen, and provide the output here

Make sure to have irecovery installed from Homebrew/MacPorts

**Comment by circularsprojects** [comment]
unable to connect to device

**Comment by LukeZGD** [comment]
[restore.zip](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14662650/restore.zip)

I'm running out of ideas at this point.. Try this one (replace restore.sh with the one in the zip file above)

**Comment by LukeZGD** [comment]
updated build (005test): https://github.com/LukeZGD/Legacy-iOS-Kit/archive/refs/heads/test.zip

please do test and see if it works this time

**Comment by circularsprojects** [comment]
I'll try it in a few hours, I don't have my 3G with me right now

**Comment by circularsprojects** [comment]
[3g005.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14686744/3g005.txt)
still same issue, this time it sent the phone to the recovery stage with the progress bar

**Comment by LukeZGD** [comment]
> it sent the phone to the recovery stage with the progress bar

This is a good sign that it's very close to getting it working now

I have pushed another update, hopefully the last one at this point

Same link above, or just replace the restore.sh with this one in the zip file: 
[restore.zip](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14691410/restore.zip)

**Comment by circularsprojects** [comment]
white screen this time, it hung on `ERROR: restore_is_current_device: can't open device with UDID 00008101-001870E601C1001E` for a few minutes before proceeding and then failing at ssh again
[3g006.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14692548/3g006.txt)

**Comment by LukeZGD** [comment]
...I'll rethink this through. Everything I have tried so far has not even worked.. I'll try to get back to you by the weekend

**Comment by LukeZGD** [comment]
Here's yet another attempt of me fixing this (007test, new branch/link): https://github.com/LukeZGD/Legacy-iOS-Kit/archive/refs/heads/test3.zip

Ditched the "Jailbreak Device" and "SSH Ramdisk" options, and instead attempted to fix the Jailbreak Option and Hacktivate Option of "Restore/Downgrade."

If you don't want to test anymore, that is fine and understandable, since this is just dragging on at this point lol. I'll probably just push this to main now even if it doesn't work on testing

**Comment by circularsprojects** [comment]
[3g007.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14711804/3g007.txt)
it's detecting the phone as the wrong model again, iPhone1,1 when it's actually iPhone1,2

**Comment by LukeZGD** [comment]
Here: [idevicerestore.zip](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14712019/idevicerestore.zip)

Go to the bin/macos folder, replace idevicerestore with the one in the zip file above, then try again

And a replacement restore.sh as well, but try the current one first before replacing: [restore.zip](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14712035/restore.zip)

I really hope that it works this time... If that still didn't work, I guess just try to restore 3.1.3...? Then if everything just doesn't work at all, I guess I'll just give up for now at this point lol

**Comment by circularsprojects** [comment]
[3g007-2.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14712102/3g007-2.txt) - replaced idevicerestore
[3g007-3.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14712105/3g007-3.txt) - replaced restore.sh as well
haven't tried to restore to 3.1.3 though

**Comment by LukeZGD** [comment]
final attempt: [idevicerestore.zip](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14715293/idevicerestore.zip)

try 3.1.3 too

**Comment by circularsprojects** [comment]
[3g007-4.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14715646/3g007-4.txt) 4.2.1 - didnt work, idevicerestore detected it as iPhone1,1
[3g007-5.txt](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14715648/3g007-5.txt) 3.1.3 - actually worked! and weirdly idevicerestore detected it correctly as iPhone1,2

**Comment by LukeZGD** [comment]
Nice! Let me know if Cydia works fine

As for 4.2.1, it looks like I will need to revert the change I made in restore.sh, so here: [restore.zip](https://github.com/LukeZGD/Legacy-iOS-Kit/files/14715691/restore.zip)

Up to you if you want to try the above or not

**Comment by LukeZGD** [comment]
here's yet another test build for attempting to fix 4.1 and 4.2.1 (008test): https://github.com/LukeZGD/Legacy-iOS-Kit/archive/refs/heads/test4.zip

Please test and let me know if it works or not. If it doesn't work, I guess it's fine since it seems that 3.1.3 works...

**Comment by LukeZGD** [comment]
Pushed changes to main now
